project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications.helper.inc

```diff
@@ -74,19 +74,49 @@ function _c4m_user_notifications_message_subscribe_get_subscribers(Message $mess
       $group = NULL;
     }
   }
-  // In this case, entity is a term, and it's notification is not supported
-  // for current MVP.
-//  else {
-//    $users =
-//  }
+
+  // Notifications for interests are "dynamic", meaning one message ID serves
+  // different term ID's.
+  // We also need to check the original node's access for the user.
+  if ($message->type === 'c4m_notify_subscribe_content_of_interest') {
+    $wrapper = entity_metadata_wrapper('message', $message);
+    $wrapper->field_taxonomy_term->set($entity_id);
+
+    // Is this a group?
+    $node = node_load($wrapper->field_node->get('nid')->value());
+
+    // Node is a group.
+    if (og_is_group_type('node', $node->type)) {
+      $group = $node;
+    }
+    // Node is group content.
+    elseif (og_is_group_content_type('node', $node->type)) {
+      $group = c4m_og_get_group_of_content($node);
+    }
+    // Site content which does not belong to a group. For example, 'article'.
+    else {
+      $group = NULL;
+    }
+  }
 
   // Filtering users and preparing final user list.
   $users = user_load_multiple(array_keys($users));
+
+  // Already subscribed users.
+  $previous_recipients = _c4m_user_notifications_get_message_unique_recipients($message->mid);
```

Comment:
I think it's best to get and use $previous_recipients before user_load_multiple() call. So that we load less users.